src/xeus_sqlite_interpreter.cpp

```diff
@@ -115,20 +115,17 @@ void interpreter::load_db(const std::vector<std::string> tokenized_code)
 
 void interpreter::create_db(const std::vector<std::string> tokenized_code)
 {
-    try
+    if (std::ofstream(tokenized_code[2].c_str()))
```

Comment:
Mmh I am not sure this is ok. You might need to close this `ofstream`. What about keeping the same code? So just
```cpp
m_bd_is_loaded = true;
m_db_path = tokenized_code[2];

// Create the file
std::ofstream(m_db_path.c_str()).close();

// Create the database
m_db = std::make_unique<SQLite::Database>(m_db_path, SQLite::OPEN_READWRITE | SQLite::OPEN_CREATE);
```

Reply:
I think `ofstream` is RAII and I need a condition for the `if`.

Reply:
Even if it's RAII, you are reading from the file before the object is deleted, so I do think you need to `close` it before reading again from it with the sqlite API.

Why do you think you need an `if` condition?

Reply:
You mean that I should leave the error to SQLite to throw?

Reply:
I can also close it later inside the if body.

Reply:
> You mean that I should leave the error to SQLite to throw?

Well that what you were doing before, no? You were catching a `std::runtime_error`, when does this `std::runtime_error` happen?

Reply:
That's true. Yeah, I'll change it.